doc improvement: Various doc fixes (#5571)
nrfconnect/sdk-nrf: samples/nrf5340/netboot/README.rst

You must use this sample as a child image of a :ref:`multi-image <ug_multi_image>` build, where MCUboot is enabled and there is a network core application.
MCUboot verifies and shares with the network core bootloader any new network core application image received through a device firmware update (DFU) transport layer, like a serial or a Bluetooth® LE connection.

> For this reason, without MCUboot, this sample does nothing other than directly launching the application.

Reviewer: ```suggestion
For this reason, without MCUboot, this sample does nothing else but directly launches the application.
```
